src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,265 @@
+        return getColor(theme, `${variantColor || 'primary'}.950`);
+    }
+  },
+  padding({ size, theme, text }) {
+    switch (size) {
+      case 'xsmall':
+        if (!text) {
+          return `${theme.spacings.xxsmall}`;
+        }
+        return `${theme.spacings.xxsmall} ${theme.spacings.small}`;
+      case 'small':
+        if (!text) {
+          return `${makePxValue(0.75)}`;
+        }
+        return `${makePxValue(0.625)} ${theme.spacings.large}`;
+      case 'medium':
+      default:
+        if (!text) {
+          return `${theme.spacings.small}`;
+        }
+        return `${theme.spacings.small} ${theme.spacings.xxlarge}`;
+      case 'large':
+        if (!text) {
+          return `${makePxValue(1.25)}`;
+        }
+        return `${makePxValue(1.25)} ${theme.spacings.xxlarge} `;
+    }
+  },
+  iconSize({ size, children }) {
+    switch (size) {
+      case 'xsmall':
+        if (!children) {
+          return 'small';
+        }
+        return 'xsmall';
+      case 'small':
+        if (!children) {
+          return 'medium';
+        }
+        return 'small';
+      case 'medium':
+      case 'large':
+      default:
+        if (!children) {
+          return 'large';
+        }
+        return 'medium';
+    }
+  },
+  fontSize({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xxsmall';
+      case 'small':
+        return 'xsmall';
+      case 'medium':
+      case 'large':
+      default:
+        return 'medium';
+    }
+  },
+  alignSelf({ align }) {
+    switch (align) {
+      case 'left':
+      default:
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+    }
+  },
+  letterSpacing({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'medium';
+      default:
+        return 'small';
+    }
+  },
+  spaceBetween({ size, iconPosition }) {
+    switch (size) {
+      case 'xsmall':
+      case 'small':
+        return iconPosition === 'left' ? [0, 0.5, 0, 0] : [0, 0, 0, 0.5];
+      case 'medium':
+      case 'large':
+      default:
+        return iconPosition === 'left' ? [0, 1, 0, 0] : [0, 0, 0, 1];
+    }
+  },
+};
 
-const StyledText = styled(Text)(
+const StyledButton = styled(TouchableHighlight)(
   (props) => `
-  color: ${props.theme.colors.tone[900]};
-  align-self: center;
-  font-size: 18px;
-  letter-spacing: ${props.theme.fonts.letterSpacing.l};
-`,
+    flex-direction: row;
+    align-items: center;
+    align-self: ${styles.alignSelf(props)};
+    background-color: ${styles.backgroundColor(props)};
+    padding: ${styles.padding(props)};
+    border-radius: ${props.theme.spacings.xxsmall};
+    ${styles.border(props) && `border: ${styles.border(props)}`};
+  `,
 );
 
-const Button = ({ onClick, children, ...rest }) => (
-  <StyledButton onPress={onClick} {...rest}>
-    <StyledText {...rest}>{children}</StyledText>
-  </StyledButton>
-);
+const Button = (props) => {
```

Comment:
```suggestion
const Button = ({
    onClick,
    children,
    variant,
    size,
    disabled,
    variantColor,
    icon,
    iconPosition,
    align,
    testID,
}) => {
```

We've been following this pattern everywhere so that we don't let any prop in that is not declared here and won't be misused by doing ```...props```